plugins/woocommerce-admin/client/products/sections/product-inventory-section.tsx

```diff
@@ -0,0 +1,81 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Card, CardBody, TextControl } from '@wordpress/components';
+import { cleanForSlug } from '@wordpress/url';
+import { getAdminLink } from '@woocommerce/settings';
+import { Link, useFormContext } from '@woocommerce/components';
+import { Product } from '@woocommerce/data';
+import { recordEvent } from '@woocommerce/tracks';
+import { useEffect } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { getTextControlProps } from './utils';
+import { ProductSectionLayout } from '../layout/product-section-layout';
+import { usePrevious } from '~/hooks/usePrevious';
+
+export const ProductInventorySection: React.FC = () => {
+	const { getInputProps, setValue, touched, values } = useFormContext< Product >();
+	const prevTouchedName = usePrevious( touched.name );
+
+	const setSkuIfEmpty = () => {
+		if ( values.sku || ! values.name?.length ) {
+			return;
+		}
+		setValue( 'sku', cleanForSlug( values.name ) );
+	};
+
+	const isNameBlurred = ! prevTouchedName && touched.name;
+	useEffect( setSkuIfEmpty, [ isNameBlurred ] );
```

Comment:
This does not handle the scenario where the user:

1. Focuses on empty name field
2. Blurs to another field (without first entering anything in name field)
3. Focuses on empty name field
4. Enters name
5. Blurs to another field

Perhaps it would be better to go back to our original implementation, which tied to the `onBlur` of the field itself. I don't love that, since it couples the two UI implementations more than ideal, but it does work better. 

Perhaps the more ideal solution needs to be addressed in the future. Maybe we need a way to hook into focus change at the Form level, so we (and extensions) could hook into whether the focus changed from one component to another at the Form level.

Or, if you changed the implementation to always update the SKU on the fly as the name value changed, keystroke by keystroke, as long as it was a new product and the user hadn't manually modified the SKU. That would be pretty slick, as long as it wasn't a performance issue.

Reply:
This was definitely a hacky solution at best.  I agree with moving back to the `onBlur`.  I'll open an issue to introduce a state handler to form context so we can detect various state changes to the form outside of specific components.

I've reverted this in https://github.com/woocommerce/woocommerce/pull/34978/commits/60da981d68f0ff9ff82742e2725d68c21b0423bc